Helpers/opent2t-translator-helper-nest/js/index.js

```diff
@@ -1,100 +1,152 @@
 'use strict';
-var https = require('follow-redirects').https;
+
+var request = require('request');
 var q = require('q');
 
-var  accessToken, deviceId, deviceCategory;
-// module exports, implementing the schema
-module.exports = {
+// wink v2 api endpoint as documented here: http://docs.winkapiv2.apiary.io/
+var apiEndpoint = 'https://developer-api.nest.com';
+
+// Internal state used to make subsequent API calls
+var bearerToken;
+
+class NestHelper {
+
+    // Init Helper: Sets the initial parameters to build our target endpoint
+    constructor(accessToken) {
+        bearerToken = accessToken;
 
-   init: function(category, id, token)
-   {
-       deviceCategory = category;
-       deviceId = id;
-       accessToken = token; //the specific token
-   },
+        console.log('Initialized Nest Helper.');
+    }
 
-    getProperty : function(name) {
-        var deferred = q.defer();   // Take a deferral
+    // Gets device details (all fields), response formatted per http:/developer.nest.com/
+    getDeviceDetailsAsync(deviceType, deviceId) {
 
+        // q will help us with returning a promise
+        var deferred = q.defer();
+
+        // build request URI
+        var requestUri = apiEndpoint + '/devices/' + deviceType + '/' + deviceId;
+
+        // Set the headers
+        var headers = {
+            'Authorization': 'Bearer ' + bearerToken
+        }
+
+        // Configure the request
         var options = {
-            protocol: 'https:',
-            host: 'developer-api.nest.com',
-            path: '/devices/'+ deviceCategory + '/' + deviceId,
-            headers: {
-                'Authorization': 'Bearer ' + accessToken
-            },
-            method: 'GET'
-        };
-
-        var httpsRequest = https.get(options, function(getRes) {
-            var body = '';
-            getRes.setEncoding('utf8');
-            getRes.on('data', function(data) {
-                body += data;
-            });
-
-            getRes.on('end', function() {
-                if (getRes.statusCode != 200) {
-                    deferred.reject(new Error("Error Code:" + getRes.statusCode));
-                }
-                else
-                {
-                    var data = JSON.parse(body);
-                    deferred.resolve(data[name]);
-                }
-            });
-
-            getRes.on('error', function(e) {
-                deferred.reject(e);
-            });
+            url: requestUri,
+            method: 'GET',
+            headers: headers,
+            followAllRedirects: true
+        }
+
+        // Start the request
+        request(options, function (error, response, body) {
+
```

Comment:
Consider using [https://github.com/request/request-promise](https://github.com/request/request-promise) for a promisified version of this API. That will make things simpler since we're using promises everywhere else.